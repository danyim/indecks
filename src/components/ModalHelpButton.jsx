import React from 'react'
import PropTypes from 'prop-types'
import ReactModal from 'react-modal'
import styles from '../styles/components/Modal.styl'

class Modal extends React.Component {
  static propTypes = {
    onRequestClose: PropTypes.func.isRequired,
    isOpen: PropTypes.bool.isRequired,
    children: PropTypes.node.isRequired,
    contentLabel: PropTypes.string.isRequired,
    onAfterOpen: PropTypes.func,
    className: PropTypes.string,
    overlayClassName: PropTypes.string
  }

  static defaultProps = {
    onAfterOpen: () => {},
    className: styles['modal-content'],
    overlayClassName: styles['modal-overlay']
  }

  render() {
    const className = `${styles['modal-content']} ${this.props.className}`
    const overlayClassName = `${styles['modal-overlay']} ${this.props
      .overlayClassName}`

    return (
      <ReactModal
        isOpen={this.props.isOpen}
        onRequestClose={this.props.onRequestClose}
        onAfterOpen={this.props.onAfterOpen}
        className={className}
        overlayClassName={overlayClassName}
        contentLabel={this.props.contentLabel}
      >
        <a
          className="close-modal"
          onClick={this.props.onRequestClose}
          role="presentation"
        >
          Close
        </a>
        {this.props.children}
      </ReactModal>
    )
  }
}

export default Modal
