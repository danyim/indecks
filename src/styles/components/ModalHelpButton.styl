@import '../_variables';

.button
  padding: 0 0.25rem
  & > i
    color: grey
    font-size: 2.0rem
    cursor: pointer

.modal
  z-index: 9000
  width: unset
  max-width: 400px
  top: unset
  left: unset
  margin-right: unset
  transform: unset
  padding-top: 4rem // Extra space to accomodate the close button
  animation: fadein .2s
  :global
    .close-modal
      position: absolute
      right: 2rem
      top: 2rem
      cursor: pointer
      color: lighten(black, 40%)
      z-index: 100

@media (max-width: cardWidth)
  .modal
    width: 95%
    height: unset
    max-height: 95%
    position: inherit
    top: 2% !important
    left: 50% !important
    transform: translate(-50%, 1%) !important
    padding: 2rem
    overflow: scroll


.overlay
  background-color: transparent

@keyframes fadein {
  from { opacity: 0; }
  to   { opacity: 1; }
}
