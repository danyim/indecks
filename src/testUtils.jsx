/*
  Not sure if we want to use this yet in our tests...
  TODO: Find a definitive structure for Jest tests and follow it.
 */
import React from 'react';
import { shallow, mount } from 'enzyme';

export const setup = (Component, props) => {
  const wrapper = shallow(<Component {...props} />)

  return {
    props,
    wrapper
  }
};

export const setupFull = (Component, props) => {
  const wrapper = mount(<Component {...props} />)

  return {
    props,
    wrapper
  }
};
