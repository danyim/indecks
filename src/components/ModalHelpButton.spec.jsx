import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow, mount, setup, setupFull } from '../testUtils' // eslint-disable-line no-unused-vars
import ModalHelpButton from './ModalHelpButton'

describe('ModalHelpButton', () => {
  let props

  beforeEach(() => {
    props = {
      children: <div />,
      style: null
    }
  })

  it('should render self and subcomponents', () => {
    const wrapper = shallow(ModalHelpButton, props)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('should toggle the help modal open/close when the icon is clicked', () => {
    const testDOM = <div className="test-dom">Test</div>
    const { wrapper } = setup(ModalHelpButton, {
      ...props,
      children: testDOM
    })

    const toggleModalIcon = wrapper.find('span[name="toggle-modal-icon"]')
    const toggle = () => {
      toggleModalIcon.simulate('click', {
        nativeEvent: {
          x: 100,
          y: 100
        }
      })
    }

    expect(wrapper.state().open).toBe(false)
    toggle()
    expect(wrapper.state().open).toBe(true)
    toggle()
    expect(wrapper.state().open).toBe(false)
  })

  it('should close the modal when close is clicked', () => {
    const { wrapper } = setup(ModalHelpButton, props)

    const toggleModalIcon = wrapper.find('span[name="toggle-modal-icon"]')
    const toggle = () => {
      toggleModalIcon.simulate('click', {
        nativeEvent: {
          x: 100,
          y: 100
        }
      })
    }
    const closeButton = wrapper.find('a.close-modal')
    toggle()
    expect(wrapper.state().open).toBe(true)
    closeButton.simulate('click')
    expect(wrapper.state().open).toBe(false)
  })

  it('should render the contents of its children', () => {
    const testDOM = <div className="test-dom">Test</div>
    const { wrapper } = setup(ModalHelpButton, {
      ...props,
      children: testDOM
    })

    const dom = wrapper.find('div.test-dom')
    expect(dom.exists()).toBe(true)
  })
})
