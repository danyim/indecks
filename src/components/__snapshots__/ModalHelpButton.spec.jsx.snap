// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalHelpButton should render self and subcomponents 1`] = `
<span>
  <span
    name="toggle-modal-icon"
    onClick={[Function]}
    role="presentation"
  >
    <i
      className="fa fa-question-circle-o"
    />
  </span>
  <Modal
    contentLabel="Help"
    isOpen={false}
    onAfterOpen={[Function]}
    onRequestClose={[Function]}
    style={
      Object {
        "left": 0,
        "top": 0,
      }
    }
  >
    <a
      className="close-modal"
      onClick={[Function]}
      role="presentation"
    >
      Close
    </a>
    <div />
  </Modal>
</span>
`;
